src/serialize-spec.ts

```diff
@@ -9,42 +10,36 @@ import {
   UInt64,
   PublicKey,
   Signature,
-  Provable,
   Undefined,
+  Bytes,
+  DynamicProof,
+  VerificationKey,
+  Struct,
 } from 'o1js';
+import { assert } from './util.ts';
 
 // Supported o1js base types
-const O1jsType = {
-  Field: 'Field',
-  Bool: 'Bool',
-  UInt8: 'UInt8',
-  UInt32: 'UInt32',
-  UInt64: 'UInt64',
-  PublicKey: 'PublicKey',
-  Signature: 'Signature',
-  Undefined: 'Undefined',
-} as const;
-
-type O1jsTypeName = (typeof O1jsType)[keyof typeof O1jsType];
-
-const supportedTypes: Record<O1jsTypeName, Provable<any>> = {
-  [O1jsType.Field]: Field,
-  [O1jsType.Bool]: Bool,
-  [O1jsType.UInt8]: UInt8,
-  [O1jsType.UInt32]: UInt32,
-  [O1jsType.UInt64]: UInt64,
-  [O1jsType.PublicKey]: PublicKey,
-  [O1jsType.Signature]: Signature,
-  [O1jsType.Undefined]: Undefined,
+const supportedTypes = {
```

Comment:
Refactor here to have only a single object where we need to list the supported types